content/kotlin/concepts/enums/enums.md

```diff
@@ -0,0 +1,228 @@
+---
+Title: 'Enums'
+Description: 'Enums are a special data type that enable a variable to be a set of predefined constants.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Enums'
+  - 'Data Types'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+**Enums** are a special data type that enable a variable to be a set of predefined constants.
+
+Kotlin enums are called enum classes and they aren't simply collections of constants, as in other languages, they also have properties, methods, etc. Each enum acts as a separate instance of the enum class, and the declaration begins with the keywords `enum class` followed by the class name.
+
+Often compared to enums corresponding to `java`, kotlin enums do not require that one knows all possible values at compile time and their declaration does not create a class behind the scenes.
+
+Beyond custom methods and properties limited to `java` enums, kotlin enums can use anonymous classes and implement interfaces.
+
+## Syntax
+
+```pseudo
+enum class <enum_class_name> {
+  constant1,
+  constant2,
+  constant3
+}
+```
+
+Enums properties and methods –
+
+Properties –
+
+- `ordinal`: This property stores the ordinal value of the constant, which is usually a zero-based index.
+- `name`: This property stores the name of the constant.
+
+Methods –
+
+- `values`: This method returns a list of all the constants defined within the enum class.
+- `valueOf`: This method returns the enum constant that matches the input string passed. If the constant, is not present in the enum, then an `IllegalArgumentException` is thrown.
+
+## Example (on properties and methods)
+
+The following example demonstrates the use of the enum class properties in Kotlin:
+
+```kotlin
+enum class GASOLINE {
+    DIESEL,
+    PREMIUM,
+    UNLEADED
+}
+fun main()
+{
+    // A simple demonstration of properties
+    for (gasoline in GASOLINE.values()) {println("${gasoline.ordinal} = ${gasoline.name}")}}
+```
+
+This prints the following output:
+
+```shell
+0 = DIESEL
+1 = PREMIUM
+2 = UNLEADED
+```
+
+and the use of methods:
+
+```kotlin
+enum class GASOLINE {
+    DIESEL,
+    PREMIUM,
+    UNLEADED
+}
+fun main()
+{
+    // A simple demonstration of methods
+    for (gasoline in GASOLINE.values()) println("${GASOLINE.valueOf("UNLEADED")}")
+}
+```
+
+the output:
+
+```shell
+UNLEADED
+UNLEADED
+UNLEADED
+```
+
+The use of when expression -
+
+One of the interesting features of kotlin enums is the complete elimination of the need to use the else clause because of the restriction on the value that a type can take.
```

Comment:
```suggestion
One of the interesting features of Kotlin enums is the complete elimination of the need to use the else clause because of the restriction on the value that a type can take.
```